### PR TITLE
Build all of the vector dir in the build process

### PR DESCRIPTION
**Pull request by dbkr**
So the whole vector/ directory can now be removed during the build
process and we don't ship random files that end up in the vector
directory.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,23 +27,27 @@
...
     "build:css": "mkdirp build && catw \"src/skins/vector/css/**/*.css\" -o build/components.css --no-watch",
     "build:compile": "babel --source-maps -d lib src",
     "build:bundle": "NODE_ENV=production webpack -p --progress",
     "build:bundle:dev": "webpack --optimize-occurence-order --progress",
     "build:electron": "build -lwm",
-    "build": "node scripts/babelcheck.js && npm run build:emojione && npm run build:css && npm run build:bundle",
+    "build": "node scripts/babelcheck.js && npm run build:res && npm run build:config && npm run build:emojione && npm run build:css && npm run build:bundle",
     "build:dev": "npm run build:emojione && npm run build:css && npm run build:bundle:dev",
     "dist": "scripts/package.sh",
+    "start:res": "parallelshell \"cpx -w \\\"{src/skins/vector/fonts,src/skins/vector/img}/**\\\" vector/\" \"cpx \\\"{res/media,res/vector-icons}/**\\\" vector/\"",
```
Comment (richvdh) [inline package.json]: the idea of nesting parallelshells makes me sad. Is there a reason these need to be separate `cpx` invocations?

Failing that the second cpx needs a -w.

Comment (dbkr) [inline package.json]: Yeah, if they're the same one, cpx copies it as a whole tree to vector/res[...] and vector/src[...] :(

**Comment by dbkr** [comment]
Yeah, possibly. OTOH, renaming the dir will break our deploy script, although maybe this is more fixable.

**Comment by dbkr** [comment]
Rewritten to use `webapp/`. Doesn't actually affect deploy script at all since we can still produce a tarball with a `vector` directory.

File: .gitignore [hunk 2]
```diff
@@ -6,11 +6,7 @@
 /lib
 /node_modules
 /packages/
-/vector/bundle.*
-/vector/emojione/
-/vector/config.json
-/vector/index.html
-/vector/olm.*
+/vector
```
Comment (richvdh) [inline .gitignore]: s/vector/webapp/

File: package.json [hunk 3]
```diff
@@ -27,23 +27,27 @@
   "matrix-react-parent": "matrix-react-sdk",
   "scripts": {
     "reskindex": "reskindex -h src/header",
-    "build:emojione": "cpx \"node_modules/emojione/assets/svg/*\" vector/emojione/svg/",
+    "build:res": "cpx \"{src/skins/vector/fonts,src/skins/vector/img}/**\" webapp/ && cpx \"{res/media,res/vector-icons}/**\" webapp/",
+    "build:config": "cpx config.json webapp/",
```
Comment (richvdh) [inline package.json]: so wasn't the point to leave `config.json` in `vector`?

Comment (dbkr) [inline package.json]: I was going to keep it in the root directory so it's more obvious to find: I don't really want a `vector` directory hanging around for the sole purpose of containing the config: people will have to move the config file but I don't think that's a big deal.

**Comment by richvdh** [comment]
LGTM modulo update to jenkins.sh

**Comment by dbkr** [comment]
Fixed - quick sanity check?

File: jenkins.sh [hunk 4]
```diff
@@ -25,11 +25,7 @@ cp -r olm/package node_modules/olm
 # run the mocha tests
 npm run test
 
-# build our artifacts; dumps them in ./vector
-npm run build:dev
-
-# gzip up ./vector
-rm vector-*.tar.gz || true # rm previous artifacts without failing if it doesn't exist
+rm packages/vector-*.tar.gz || true # rm previous artifacts without failing if it doesn't exist
```
Comment (richvdh) [inline jenkins.sh]: `package.sh` seems to build to dist/, not packages/ ?

Comment (dbkr) [inline jenkins.sh]: ah yes, because I changed that the other day.

**Pull request merged**